pyat/at/collective/beam_loading.py

```diff
@@ -193,16 +210,25 @@ def clear_history(self, ring=None):
         self._init_bl_params(current)
 
     def _init_bl_params(self, current):
-        theta = -self._vcav[1]+numpy.pi/2
-        vb = 2*current*self.Rshunt
-        a = self.Voltage*numpy.cos(theta-self._phis)
-        b = self.Voltage*numpy.sin(theta-self._phis)-vb*numpy.cos(theta)
-        psi = numpy.arcsin(b/numpy.sqrt(a**2+b**2))
-        vgen = self.Voltage*numpy.cos(psi) + \
-            vb*numpy.cos(psi)*numpy.sin(self._phis)
-        self._vgen = numpy.array([vgen, psi])
+        if (self._cavitymode == 1) and (current > 0.0):
+            theta = -self._vcav[1]+numpy.pi/2
+            vb = 2*current*self.Rshunt
+            a = self.Voltage*numpy.cos(theta-self._phis)
+            b = self.Voltage*numpy.sin(theta-self._phis)-vb*numpy.cos(theta)
+            psi = numpy.arcsin(b/numpy.sqrt(a**2+b**2))
+            vgen = self.Voltage*numpy.cos(psi) + \
+                vb*numpy.cos(psi)*numpy.sin(self._phis)
+
```

Comment:
I am not very keen on these extra empty lines....

Reply:
done